crates/ty_python_semantic/src/types/ide_support.rs

```diff
@@ -981,4 +992,207 @@ mod resolve_definition {
 
         definitions
     }
+
+    #[tracing::instrument(skip_all)]
+    pub fn map_stub_definition<'db>(
+        db: &'db dyn Db,
+        def: &ResolvedDefinition<'db>,
+    ) -> Option<Vec<ResolvedDefinition<'db>>> {
+        trace!("Stub mapping definition...");
+        // If the file isn't a stub, this is presumably the real definition
+        let stub_file = def.file(db);
+        if !stub_file.is_stub(db) {
+            trace!("File isn't a stub, no stub mapping to do");
+            return None;
+        }
+
+        // It's definitely a stub, so now rerun module resolution but with stubs disabled.
+        let stub_module = file_to_module(db, stub_file)?;
+        trace!("Found stub module: {}", stub_module.name());
+        let real_module = resolve_real_module(db, stub_module.name())?;
+        trace!("Found real module: {}", real_module.name());
```

Comment:
I'm choosing to take the insinuation that stubs are somehow "fake" or "inauthentic" as a personal affront, as a typeshed maintainer 😆

Reply:
Only in the artifice can we find perfection